### ISSUE TITLE
[SDL2] Reinit devices with `SDL_HIDAPI_LIBUSB=ON`

### ISSUE THREAD

**Issue by sr-tream** [libsdl-org/SDL#8373]
With flag `SDL_HIDAPI_LIBUSB=ON` SDL reinit my devices on launch SDL apps. This reinit stops control LEDs via OpenRGB.

Affected device: Logitech G815
SDL2 version: 2.28.4

**Comment by slouken** [comment]
What OS are you running?

**Comment by sr-tream** [comment]
OS: Linux (Archlinux)

SDL2 builded from git

**Comment by slouken** [comment]
Can you check to see if this fixes it for you?
```diff
diff --git a/src/joystick/hidapi/SDL_hidapijoystick.c b/src/joystick/hidapi/SDL_hidapijoystick.c
index a3c6872b0..11fe5e6db 100644
--- a/src/joystick/hidapi/SDL_hidapijoystick.c
+++ b/src/joystick/hidapi/SDL_hidapijoystick.c
@@ -152,8 +152,12 @@ SDL_bool HIDAPI_SupportsPlaystationDetection(Uint16 vendor, Uint16 product)
     case USB_VENDOR_HORI:
         return SDL_TRUE;
     case USB_VENDOR_LOGITECH:
-        /* Most Logitech devices are fine with this, but the F310 will lock up */
         if (product == USB_PRODUCT_LOGITECH_F310) {
+            /* Most Logitech devices are fine with this, but the F310 will lock up */
+            return SDL_FALSE;
+        }
+        if (product == 0xc33f) {
+            /* The G815 keyboard will reset the LEDs */
             return SDL_FALSE;
         }
         return SDL_TRUE;

```

**Comment by sr-tream** [comment]
> Can you check to see if this fixes it for you?
> 
> ```diff
> diff --git a/src/joystick/hidapi/SDL_hidapijoystick.c b/src/joystick/hidapi/SDL_hidapijoystick.c
> index a3c6872b0..11fe5e6db 100644
> --- a/src/joystick/hidapi/SDL_hidapijoystick.c
> +++ b/src/joystick/hidapi/SDL_hidapijoystick.c
> @@ -152,8 +152,12 @@ SDL_bool HIDAPI_SupportsPlaystationDetection(Uint16 vendor, Uint16 product)
>      case USB_VENDOR_HORI:
>          return SDL_TRUE;
>      case USB_VENDOR_LOGITECH:
> -        /* Most Logitech devices are fine with this, but the F310 will lock up */
>          if (product == USB_PRODUCT_LOGITECH_F310) {
> +            /* Most Logitech devices are fine with this, but the F310 will lock up */
> +            return SDL_FALSE;
> +        }
> +        if (product == 0xc33f) {
> +            /* The G815 keyboard will reset the LEDs */
>              return SDL_FALSE;
>          }
>          return SDL_TRUE;
> ```

Yes, it fixes

**Comment by slouken** [comment]
Fixed for the SDL 2.30 release, thanks!
